compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/TextFieldPointerModifier.uikit.kt

```diff
@@ -0,0 +1,210 @@
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.text
+
+import androidx.compose.foundation.gestures.detectDragGesturesAfterLongPress
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.text.selection.SelectionAdjustment
+import androidx.compose.foundation.text.selection.TextFieldSelectionManager
+import androidx.compose.foundation.text.selection.getTextFieldSelection
+import androidx.compose.foundation.text.selection.isSelectionHandleInVisibleBound
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.hapticfeedback.HapticFeedbackType
+import androidx.compose.ui.input.pointer.pointerHoverIcon
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TextFieldValue
+
+internal actual fun Modifier.textFieldPointer(
+    manager: TextFieldSelectionManager,
+    enabled: Boolean,
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping
+): Modifier = if (enabled) {
+    if (isInTouchMode) {
+        val selectionModifier = getSelectionModifier(manager)
+        val tapHandlerModifier = getTapHandlerModifier(
+            interactionSource,
+            state,
+            focusRequester,
+            readOnly,
+            offsetMapping,
+            manager
+        )
+        tapHandlerModifier
+            .then(selectionModifier)
+            .pointerHoverIcon(textPointerIcon)
+    } else {
+        Modifier
+            .mouseDragGestureDetector(
+                observer = manager.mouseSelectionObserver,
+                enabled = enabled
+            )
+            .pointerHoverIcon(textPointerIcon)
+    }
+} else {
+    Modifier
+}
+
+@OptIn(InternalFoundationTextApi::class)
+private fun getTapHandlerModifier(
+    interactionSource: MutableInteractionSource?,
+    state: TextFieldState,
+    focusRequester: FocusRequester,
+    readOnly: Boolean,
+    offsetMapping: OffsetMapping,
+    manager: TextFieldSelectionManager
+) = Modifier.pointerInput(interactionSource) {
+    /*
+    We need to move tap recognizer here from selection modifier (as it is in common)
+    because otherwise we have onDoubleTap call and onTap call, and onDoubleTap will execute before onTap.
+     */
+    detectTapGestures(
+        onDoubleTap = { touchPointOffset ->
+            tapTextFieldToFocus(
+                state,
+                focusRequester,
+                !readOnly
+            )
+            manager.doDoubleTapSelection(touchPointOffset)
+        },
+        onTap = { touchPointOffset ->
```

Comment:
```suggestion
        onPress = { touchPointOffset ->
```
`onTap` is called with visible delay (to detect double tap). We should use `onPress` instead (as in `tapPressTextFieldModifier`).

Reply:
Thanks, I'll remember it. But why does it exist then?

Reply:
In case, we need to either handle a single tap or a double tap.

When we handle press - we always handle it, even if there is double tap after it.